x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsPersister.java

```diff
@@ -328,12 +328,16 @@ public void persistQuantiles(Quantiles quantiles, Supplier<Boolean> shouldRetry)
             shouldRetry,
             retryMessage -> logger.debug("[{}] {} {}", jobId, quantilesDocId, retryMessage)
         );
-        String indexOrAlias = searchResponse.getHits().getHits().length > 0
-            ? searchResponse.getHits().getHits()[0].getIndex()
-            : AnomalyDetectorsIndex.jobStateIndexWriteAlias();
+        try {
+            String indexOrAlias = searchResponse.getHits().getHits().length > 0
+                ? searchResponse.getHits().getHits()[0].getIndex()
+                : AnomalyDetectorsIndex.jobStateIndexWriteAlias();
 
-        Persistable persistable = new Persistable(indexOrAlias, quantiles.getJobId(), quantiles, quantilesDocId);
-        persistable.persist(shouldRetry, AnomalyDetectorsIndex.jobStateIndexWriteAlias().equals(indexOrAlias));
+            Persistable persistable = new Persistable(indexOrAlias, quantiles.getJobId(), quantiles, quantilesDocId);
+            persistable.persist(shouldRetry, AnomalyDetectorsIndex.jobStateIndexWriteAlias().equals(indexOrAlias));
+        } finally {
+            searchResponse.decRef();
+        }
     }
```

Comment:
`persistable.persist(shouldRetry, AnomalyDetectorsIndex.jobStateIndexWriteAlias().equals(indexOrAlias));` can retry many times and unnecessarily increase the time we keep `searchResponse` referenced. We only need `indexOrAlias` from the response, so I think we grab that, and then decRef before persisting.